internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -526,58 +524,64 @@ func (s *BackendService) convertALBListener() (*template.ALBListener, error) {
 		return nil, nil
 	}
 	var rules []template.ALBListenerRule
-	// build listener rule config from primary rule config from manifest.
-	rule, err := routingRuleConfigConverter{
-		rule:         rrConfig.Main,
-		manifest:     s.manifest,
-		httpsEnabled: s.httpsEnabled,
-	}.convert()
-	if err != nil {
-		return nil, err
-	}
-	rules = append(rules, *rule)
-	hostedZoneAliases, err := convertHostedZone(rrConfig.Main.Alias, rrConfig.Main.HostedZone)
-	if err != nil {
-		return nil, err
+	var hostedZoneAliases template.AliasesForHostedZone
+	for _, routingRule := range rrConfig.RoutingRules() {
+		rule, err := routingRuleConfigConverter{
+			rule:            routingRule,
+			manifest:        s.manifest,
+			httpsEnabled:    s.httpsEnabled,
+			redirectToHTTPS: s.httpsEnabled,
+		}.convert()
+		if err != nil {
+			return nil, err
+		}
+		rules = append(rules, *rule)
+		hostedZoneAliases, err = convertHostedZone(rrConfig.Main.Alias, rrConfig.Main.HostedZone)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	// TODO: @pbhingre build listener rule config from additional rules from manifest.
-
 	return &template.ALBListener{
 		Rules:             rules,
-		RedirectToHTTPS:   s.httpsEnabled,
 		IsHTTPS:           s.httpsEnabled,
 		MainContainerPort: s.manifest.MainContainerPort(),
 		HostedZoneAliases: hostedZoneAliases,
 	}, nil
 }
 
 type loadBalancerTargeter interface {
-	HTTPLoadBalancerTarget() (string, string, error)
 	MainContainerPort() string
+	ExposedPorts() (manifest.ExposedPortsIndex, error)
 }
 
 type routingRuleConfigConverter struct {
-	rule         manifest.RoutingRule
-	manifest     loadBalancerTargeter
-	httpsEnabled bool
+	rule            manifest.RoutingRule
+	manifest        loadBalancerTargeter
+	httpsEnabled    bool
+	redirectToHTTPS bool
 }
 
 func (conv routingRuleConfigConverter) convert() (*template.ALBListenerRule, error) {
 	var aliases []string
 	var err error
+
+	exposedPorts, err := conv.manifest.ExposedPorts()
```

Comment:
why do we call `ExposedPorts` outside rather than within `Target()`?

Reply:
Because we are keeping `Target()` on listener rule level and not manifest level as target is listener rule level entity. Does that make sense?

Reply:
In that case would it make more sense to revert rule and exposedPorts? Like `exposedPorts.Target(conv.rule)`? As `exposedPorts` is actually the one with more info.

Reply:
As per the offline discussion keeping it as is for readability and consistency purpose with NLB.